File: back_end/VehicleTracker/src/main/java/com/app/model/Vehicle.java
package com.app.model;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Vehicle {
    @Id
    @Column(length = 30)
    private String vechicleNumber;
    @Column(length = 30 , nullable = false)
    private String vehicleType;
    @Column(columnDefinition = "int default 0")
    private Integer challan_Exp;   // expenditure
    @Column(columnDefinition = "int default 0")
    private Integer vehicle_Exp;

    @ManyToOne(optional = false)
    @JoinColumn(name = "user")
    private User user;
}

File: back_end_dotnet/vehicleTracker_dotnet/Model/VTModels/Vehicle.cs
using System.ComponentModel.DataAnnotations;

namespace VehicleTracker.Models;

public class Vehicle
{
    [Key]
    [MaxLength(30)]
    public string VechicleNumber { get; set; }


    [Required, MaxLength(30)]
    public string VehicleType { get; set; }


    public int Challan_Exp { get; set; } = 0;


    public int Vehicle_Exp { get; set; } = 0;


    [Required]
    public User User { get; set; }


}
